plugins/genesys/fu-genesys-usbhub-device.c

```diff
@@ -1026,6 +1026,35 @@ fu_genesys_tsdigit_value(gchar c)
 	return g_ascii_digit_value(c);
 }
 
+static gboolean
+fu_genesys_usbhub_device_validate_token(FuGenesysUsbhubDevice *self, GError **error)
+{
+	const gchar *valid_tokens[] = {"GL 3.0 Hub", "GL 3.1 Hub", NULL};
+	g_autofree gchar *token = NULL;
+	g_autoptr(GBytes) token_blob = NULL;
+
+	/* get 0x80 string descriptor */
+	token_blob = g_usb_device_get_string_descriptor_bytes_full(
+	    fu_usb_device_get_dev(FU_USB_DEVICE(self)),
+	    0x80,
+	    G_USB_DEVICE_LANGID_ENGLISH_UNITED_STATES,
+	    64,
+	    error);
+	if (token_blob == NULL)
```

Comment:
```
fwupdtool get-devices --plugins genesys
Loading…                 [********                               ]10:09:26.190 FuEngine             failed to add device usb:01:00:02:07: USB error on device 05e3:0610 : Pipe error [-9]
Loading…                 [***********                            ]10:09:26.202 FuEngine             no GUIDs for device USB HID DEVICE [92190d8f48c2ca72b37e3c3c798a4c6405d8c543]
10:09:26.202 FuEngine             no GUIDs for device USB HID DEVICE [92190d8f48c2ca72b37e3c3c798a4c6405d8c543]
Loading…                 [************************************** ]
WARNING: This package has not been validated, it may not work properly.
Intel(R) Client Systems NUC8i5BEH
│
└─USB2.1 Hub:
      Device ID:          301046452a49d84af6356d23e43a684b8f10660f
      Current version:    89.4
      Vendor:             GenesysLogic (USB:0x05E3)
      Install Duration:   9 seconds
      GUIDs:              57fc80bd-e53a-573d-8d76-728a2f787adf ← USB\VID_05E3&PID_0630&IC_352510
                          78d2d4f5-98de-5a3e-894d-d0487ae51100 ← USB\VID_05E3&PID_0630&IC_352510&BONDING_F8
                          3c1cd9dd-9c16-5e39-a876-cb3916edca8e ← USB\VID_05E3&PID_0630&VENDOR_GENESYSLOGIC&IC_352510&BONDING_F8&PORTNUM_49&VENDORSUP_3E788A7C-9923-544D-AB03-A37167E44AFA
                          96b4c772-f531-55dc-87dc-11ac566bc488 ← USB\VID_05E3&PID_0630&PROJECT_BE0F5C39-41D5-55D9-88F0-681418FDEF19
      Device Flags:       • Updatable
                          • Cryptographic hash verification is available
                          • Unsigned Payload

```

Reply:
> It's kept error here, should we set another error?

IIUC, I think we're aborting anytime the function fails -- i.e. it's okay.